static/js/TextColumnBanner.jsx

```diff
@@ -0,0 +1,132 @@
+import React, {useState} from "react";
+import Sefaria  from './sefaria/sefaria';
+import $  from './sefaria/sefariaJquery';
+import {
+    CloseButton, InterfaceText, EnglishText, HebrewText
+} from './Misc';
+
+const cookie = Sefaria._inBrowser ? $.cookie : Sefaria.util.cookie;
+const { translation_language_preference_suggestion } = Sefaria;
+
+export const TextColumnBannerChooser = ({ setTranslationLanguagePreference, openTranslations, transCallToActionApplies }) => {
+    const [bannerAccepted, setBannerAccepted] = useState(false);
```

Comment:
I'm not sure i understand the logic here. accepted only seems to be used for one kind of banner and it seems to make sense to use more generally if its out here in the choose component?

Reply:
This works, but if more banners get added in the future, I'm concerned the order of operations here is going to get more and more murky and we have states that are only relevant for one sub component